### PR TITLE
fix(LLM/LLC): Fiat currencies fetch logic rework to fix issues

### PR DESCRIPTION
**Pull request by mle-gall**
### 📝 Description

Updated fiat currencies call and logic on LLM which prevented appropriate data being displayed

### ❓ Context

- **JIRA or GitHub link**:  [PROD-5432]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[PROD-5432]: https://ledgerhq.atlassian.net/browse/PROD-5432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/coin-framework/src/currencies/support.ts [hunk 1]
```diff
@@ -104,7 +99,7 @@ export async function listSupportedFiats(): Promise<FiatCurrency[]> {
     } catch (error) {
       // Handle initialization error
       log("debug", `Failed to initialize userSupportedFiats. Error Message: ${error}`);
-      return [];
+      return userSupportedFiats || [];
```
Comment (mcayuelas-ledger) [inline libs/coin-framework/src/currencies/support.ts:102]: If we enter catch, it seems that `userSupportedFiats` are not loader right?

File: apps/ledger-live-mobile/src/context/LedgerStore.tsx [hunk 2]
```diff
@@ -70,24 +70,26 @@ export default class LedgerStoreProvider extends Component<
...
-      const supportedCounterValues = [...supportedFiats, ...possibleIntermediaries]
-        .map(currency => ({
-          value: currency.ticker,
-          ticker: currency.ticker,
-          label: `${currency.name} - ${currency.ticker}`,
-          currency,
-        }))
-        .sort((a, b) => (a.currency.name < b.currency.name ? -1 : 1));
-      return supportedCounterValues;
+      const supportedCounterValues = await listSupportedFiats().then(res =>
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/context/LedgerStore.tsx:73]: Why do we mix async/await + Promise.then?

File: libs/coin-framework/src/currencies/support.ts [hunk 3]
```diff
@@ -52,22 +52,17 @@ export const locallySupportedFiats = [
...
-      // This makes sure we only keep the elements supported in our API and that are available for getFiatCurrencyByTicker
-      supportedTokens = idsToUpper.filter(token => locallySupportedFiats.includes(token));
-    }
-    userSupportedFiats = supportedTokens.map(id => {
-      return getFiatCurrencyByTicker(id);
-    });
-  } catch (error) {
-    throw new Error(`Failed to get supported Fiats. Error Message: ${error}`);
+    // This makes sure we only keep the elements supported in our API and that are available for getFiatCurrencyByTicker
+    supportedTokens = idsToUpper.filter(token => locallySupportedFiats.includes(token));
```
Comment (KVNLS) [inline libs/coin-framework/src/currencies/support.ts:61]: I think that we can iterate only once on ids here.

**Review by KVNLS — CHANGES_REQUESTED** [review]

**Review by gre — COMMENTED** [review]
ok for now, just a side note here: i think in future we will want to remove the state "userSupportedFiats" that needs to live in the redux side and not hidden in a module (the coin-framework), but that will require more than a simple fix

File: libs/coin-framework/src/currencies/support.ts [hunk 4]
```diff
@@ -52,22 +52,17 @@ export const locallySupportedFiats = [
...
-    const ids = await fetchSupportedFiatsTokens();
-    let supportedTokens = locallySupportedFiats;
-
-    if (ids.length) {
-      const idsToUpper = ids?.map(id => id.toUpperCase());
-
-      // This makes sure we only keep the elements supported in our API and that are available for getFiatCurrencyByTicker
-      supportedTokens = idsToUpper.filter(token => locallySupportedFiats.includes(token));
+  const ids = await fetchSupportedFiatsTokens();
+  const supportedTokens = locallySupportedFiats;
```
Comment (KVNLS) [inline libs/coin-framework/src/currencies/support.ts:56]: We don't need this instruction. As we copy the reference when we do `supportedTokens.push(token);` we append to locallySupportedFiats too.

File: apps/ledger-live-mobile/src/context/LedgerStore.tsx [hunk 5]
```diff
@@ -70,24 +70,29 @@ export default class LedgerStoreProvider extends Component<
...
         .map(currency => ({
           value: currency.ticker,
           ticker: currency.ticker,
           label: `${currency.name} - ${currency.ticker}`,
           currency,
         }))
         .sort((a, b) => (a.currency.name < b.currency.name ? -1 : 1));
-      return supportedCounterValues;
+
+      if (this.props.store && this.props.store.dispatch) {
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/context/LedgerStore.tsx:83]: You can replace this check with optional chaining.

File: apps/ledger-live-mobile/src/context/LedgerStore.tsx [hunk 6]
```diff
@@ -70,24 +70,29 @@ export default class LedgerStoreProvider extends Component<
     const possibleIntermediaries = [bitcoin, ethereum];
 
     const getsupportedCountervalues = async () => {
-      const supportedFiats = await listSupportedFiats();
-      const supportedCounterValues = [...supportedFiats, ...possibleIntermediaries]
+      const res = await listSupportedFiats();
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/context/LedgerStore.tsx:73]: I think that `supportedFiats` was a better variable name than `res`, is there a reason for this change that I don't see?

**Review by KVNLS — CHANGES_REQUESTED** [review]
The fix works well! I added few comments on the code

File: libs/coin-framework/src/currencies/support.ts (hunk 4 above)
Comment (mle-gall) [reply libs/coin-framework/src/currencies/support.ts]: Modified the function for a clearer code.

**Review by haammar-ledger — APPROVED** [review]

**Review by sshmaxime — APPROVED** [review]

**Pull request merged**